src/main/java/com/datadoghq/datadog_lambda_java/MetricWriter.java

```diff
@@ -39,20 +39,30 @@ public void flush(){}
 
 class ExtensionMetricWriter extends MetricWriter{
 
-    private StatsDClient client;
-
-    public ExtensionMetricWriter() {
-        try {
-            this.client = new NonBlockingStatsDClientBuilder()
-                    .prefix("")
-                    .hostname("127.0.0.1")
-                    .port(8125)
-                    .enableTelemetry(false)
-                    .telemetryFlushInterval(0)
-                    .build();
-        } catch (Exception e) {
-            DDLogger.getLoggerImpl().error("Could not create StatsDClient " + e.getMessage());
-            this.client = null;
+    private static StatsDClient client;
+    private static ExtensionMetricWriter emw;
+
+    public static ExtensionMetricWriter GetInstance(){
+        if (null == emw){
+            emw = new ExtensionMetricWriter();
```

Comment:
Good idea to use a singleton instance!

Reply:
If we're being perfectly honest, I got the idea from @junkfactory in the issue they opened.

> Initializing MetricWriter could naively be in a static initializer but that's just me.